File: libs/aoc_runner/python/aoc_runner.py
import argparse
import sys


def aoc_runner(prechecks, sol1, sol2, data) -> None:
    parser = argparse.ArgumentParser(description="aoc-runner-py")

    parser.add_argument("-s", "--samples", action="store_true", help="run prechecks")
    parser.add_argument("-a", "--all", action="store_true", help="run everything")
    parser.add_argument(
        "-p", "--part", type=int, choices=[1, 2], help="run specific part [1,2]"
    )
    args = parser.parse_args()

    if len(sys.argv) - 1 == 0:
        args.samples = True

    if args.all:
        args.samples = True
        args.part = None

    args.part_1 = False
    args.part_2 = False
    match args.part:
        case 1:
            args.part_1 = True
        case 2:
            args.part_2 = True
        case _:
            args.part_1 = True
            args.part_2 = True

    if args.samples:
        prechecks()
        print("checks done")

    if args.part_1:
        sol1(data)

    if args.part_2:
        sol2(data)

File: libs/aoc_runner/go/aoc_runner.go
package aoc_runner

import (
	"flag"
)

func Run(samples func(), part1 func([]string), part2 func([]string), input []string) {
	var part int
	flag.IntVar(&part, "part", 0, "part")
	flag.Parse()

	switch part  {
	case 1:
		part1(input);
		break
	case 2:
		part2(input);
		break
	default:
		samples();
		part1(input);
		part2(input);
		break
	}
}
